### PR TITLE
[MTK] Add support for Llama 3.2 and code updates to align with current ET API for dynamic dim

### PR DESCRIPTION
**Pull request by neuropilot-captain**
This PR contains the following modifications/udpates:

- Add Llama 3.2 related files
- Fix bug for tie word embeddings weight naming
- Exclude embedding .bin file during weight checking in sanity checks
- Get Embedding layer before loading of main model if calibration dataset provided to prevent tied word embedding weight name from being removed from state dict during main model weight loading
- Add keep_in_memory during dataset mapping function to prevent disk out of space issue

### CODE REVIEW

File: examples/mediatek/model_export_scripts/llama.py [hunk 1]
```diff
@@ -460,6 +462,7 @@ def main():
                 "eos_token_id_tensor": torch.tensor(tokenizer.eos_token_id),
                 "response_cap": args.response_cap,
             },
+            keep_in_memory=True
```
Comment (cccclai) [inline examples/mediatek/model_export_scripts/llama.py:465]: Mind sharing what is keep in memory?

File: examples/mediatek/models/llm_models/modeling_common.py [hunk 2]
```diff
@@ -757,9 +757,9 @@ def get_example_inputs(
             nt = Dim("num_token", max=num_token)
             cache_dims = tuple(({} for _ in range(2 * self.num_blocks)))
             dynamic_shapes = (
```
Comment (cccclai) [inline examples/mediatek/models/llm_models/modeling_common.py:759]: I think I may ask in the previous PR - what does `dynamic_shape` do? Probably good to add a comment to explain

File: examples/mediatek/model_export_scripts/llama.py [hunk 3]
```diff
@@ -437,7 +440,6 @@ def main():
     cal_dataset = None
     if args.dataset is not None:
         cal_dataset = load_dataset("text", data_files=args.dataset, split="train")
-        embedding_layer = get_embedding_layer(config, weight_dir, state_dict)
```
Comment (cccclai) [inline examples/mediatek/model_export_scripts/llama.py:440]: Any specific reason to remove this line?

**Review by cccclai — APPROVED** [review]
Most nit comments, thank you for adding support for 3.2

**Comment by cmodi-meta** [comment]
Thanks @neuropilot-captain! do we also need to make to the `examples/mediatek/shell_scripts/export_llama.sh` file to include Llama 3.2 1B and 3B model in the if-else conditions?

**Comment by cccclai** [comment]
There are some lint errors. Could you send a fix?

File: examples/mediatek/model_export_scripts/llama.py (hunk 1 above)
Comment (neuropilot-captain) [reply examples/mediatek/model_export_scripts/llama.py:465]: Hi @cccclai, keep in memory stores the dataset in RAM instead of caching it. It was a temporary workaround for an OSError I encountered on my end. I have since resolved the issue and will remove this argument in the next commit.

File: examples/mediatek/models/llm_models/modeling_common.py (hunk 2 above)
Comment (neuropilot-captain) [reply examples/mediatek/models/llm_models/modeling_common.py:759]: Hi @cccclai, dynamic_shape is passed in to torch.export.export_for_training to indicate that the input shapes during calibration for some of the inputs may be different. After calibration, the dynamic_shape argument is not needed

File: examples/mediatek/model_export_scripts/llama.py (hunk 3 above)
Comment (neuropilot-captain) [reply examples/mediatek/model_export_scripts/llama.py:440]: Oh I didn't remove it. I shifted it above to line 422. The reason being that during chunk.load_weights (line 437), the weights are popped from the original state_dict and for the tie word embedding True case, this would mean that the embedding weights are not present in the state_dict anymore by the time we get the embedding layer, hence I shifted it up

**Comment by neuropilot-captain** [comment]
> Thanks @neuropilot-captain! do we also need to make to the `examples/mediatek/shell_scripts/export_llama.sh` file to include Llama 3.2 1B and 3B model in the if-else conditions?

Hi @cmodi-meta, yes the script would need to be modified to include Llama 3.2 1B and 3B in the if else. I can modify the script and update it in the next commit
